language_formatters_pre_commit_hooks/pretty_format_yaml.py

```diff
@@ -42,31 +43,38 @@ def pretty_format_yaml(argv=None):
     # Prevent ruamel.yaml to wrap yaml lines
     yaml.width = maxsize
 
+    separator = '---\n'
+
     for yaml_file in set(args.filenames):
         with open(yaml_file) as f:
             string_content = ''.join(f.readlines())
 
-        try:
-            content = list(yaml.load_all(string_content))
-            pretty_content = StringIO()
-
-            if len(content) == 1:
-                if not isinstance(content[0], (list, dict)):
-                    # skip files containing primitive types (unstructured text)
-                    continue
+        separator_pattern = '^--- ?.*\\n'
```

Comment:
You're right about the standard. I decided to be more flexible because of examples like the one [here](https://yaml.org/start.html). Nonetheless, it makes no difference to me. I'm happy to change it.